lib/interfaces/swagger-custom-options.interface.ts

```diff
@@ -16,45 +12,3 @@ export interface ExpressSwaggerCustomOptions
   url?: string;
   urls?: Record<'url' | 'name', string>[];
 }
-
-export interface FastifySwaggerCustomOptions
```

Comment:
Could we add an extra, temporary compatibility layer that would allow us to merge this PR without publishing a new major release? A layer that would still let library consumers pass `FastifySwaggerCustomOptions` etc and would convert them (options) to whatever interface we require them to pass now. We could remove this layer later on in the next release. Let me know if it's feasible 🙌  @flamewow

Reply:
we can have a compatibility layer for FastifySwaggerCustomOptions.uiConfig

but for  
` staticCSP?: boolean | string | Record<string, string | string[]>;
  transformStaticCSP?: (header: string) => string;
  uiHooks?: {
    onRequest?: Function;
    preHandler?: Function;
  };`
  there's no equivalent functionality in swagger-ui-express (don't know if it's even needed tbh)

Do u think adding partial compatibility layer, i.e. old fastify nestjs/swagger use cases won't break and those features (as uiHooks) just won't work is ok?
Or it worth making it a major upgrade forcing everyone to consciously migrate their swagger initialization?

Reply:
@flamewow I've seen codebases where `uiHooks` were used for Basic authentication (to access swagger docs). Maybe we can somehow pass them down? Could you take a look at `fastify-swagger` and try to mimic some of its functionality (totally understandable if having 100% parity isn't doable)?

Reply:
@kamilmysliwiec tried to make uiHooks work with the new approach, unfortunately no success here.
In order to keep feature compatibility (uiHooks specifically) I'll need to copy good chunk of fastify-swagger code (fastify routing + plugin registration etc) which makes the whole point of the PR questionable.

Added backward compatibility layer for uiConfig.

Reply:
> I'll need to copy good chunk of fastify-swagger code (fastify routing + plugin registration etc)

I think that's _fine_(?). How much code would we have to move? Just double-checked and it seems that the only thing we'd need is this https://github.com/fastify/fastify-swagger/blob/master/lib/routes.js#L89 (adding `hooks` to the route _if registered_), but correct me if I'm wrong .

Perhaps we could just add another condition in the `serveDocuments` and if platform === fastify then grab Fastify adapter and use `.route()` to register hooks? Obviously it would make this method less "platform-agnostic" (currently it only uses `HttpAdapter` which is great) but at least would let folks keep using `uiHooks` which seems to be crucial

Reply:
@kamilmysliwiec updated (included serveDocumentsFastify function to handle documents in fastify case, supports uiHooks)
Added sample code to showcase uiHooks and ability to set custom CSS (in fastify case).

P.S. it is not as bad as I anticipated, guess I was too focused on making it truly universal )